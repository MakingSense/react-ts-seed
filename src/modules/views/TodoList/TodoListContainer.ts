import { connect } from 'react-redux';

import { IRootState } from '../../state-mgmt/rootState';
import { todoState } from '../../state-mgmt/todo';
import TodoList from './TodoList';

export const mapStateToProps = (state: IRootState) => ({
  todoMap: state.todo.todoMap
});

export const mapDispatchToProps = dispatch => ({
  fetchTodoList: () => dispatch(todoState.actions.fetchStart())
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
