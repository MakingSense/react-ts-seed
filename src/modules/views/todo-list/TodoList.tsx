import * as React from 'react';

import { GeneralModel, TodoModel } from '../../models';

export interface ILoginProps {
  todoMap: GeneralModel.IEntityMap<TodoModel.ITodo>;
  fetchTodoList: () => void;
}

export interface ILoginState { }

export default class Login extends React.PureComponent<ILoginProps, ILoginState> {
  public componentDidMount() {
    const { fetchTodoList } = this.props;
    fetchTodoList();
  }

  public render() {
    const { todoMap } = this.props;
    const todoList = Object.values(todoMap);
    return (
      <div className="container" key="TodoList">
        <table className="table">
          <thead>
            <tr>
              <th>ID</th>
              <th>name</th>
              <th>description</th>
            </tr>
          </thead>
          <tbody>
            {
              todoList.map(todo => (
                <tr key={todo.id}>
                  <td>{todo.id}</td>
                  <td>{todo.name}</td>
                  <td>{todo.description}</td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>
    );
  }
}
