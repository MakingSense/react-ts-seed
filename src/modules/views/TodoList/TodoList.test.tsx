import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { getTodo_1 } from '../../../test/entities';
import TodoList, { ITodoListProps } from './TodoList';

describe('TodoList Component', () => {
  let wrapper: RenderResult;
  let props: ITodoListProps;

  beforeEach(() => {
    props = {
      todoMap: { [getTodo_1().id]: getTodo_1() },
      fetchTodoList: jest.fn()
    };
    wrapper = render(<TodoList {...props} />);
  });

  it('should render', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
