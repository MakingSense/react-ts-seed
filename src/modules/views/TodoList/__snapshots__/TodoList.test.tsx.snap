// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component should render 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              name
            </th>
            <th>
              description
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              some todo
            </td>
            <td>
              some desc
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
